src/pybind/mgr/dashboard_v2/models/nodb.py

```diff
@@ -0,0 +1,897 @@
+def subclass(name, parents, class_dict=None):
+    if class_dict is None:
+        class_dict = {}
+    return type(name, tuple(parents), class_dict)
+
+class NodbOptions(object):
+    def __init__(self, meta):
+        self.meta = meta
+        self.pk = None
+        self.fields = []
+        self.unique_together = None
+        self.object_name = None
+        self.model = None
+        self.model_name = None
+
+    def _prepare(self, model):
+        if self.pk is None:
+            auto = IntegerField(verbose_name='ID', primary_key=True)
+            model.add_to_class('id', auto)
+
+    def contribute_to_class(self, cls, name):
+
+        cls._meta = self
+        self.model = cls
+        # First, construct the default values for these options.
+        self.object_name = cls.__name__
+        self.model_name = self.object_name.lower()
+
+        # Next, apply any overridden values from 'class Meta'.
+        if self.meta:
+            meta_attrs = self.meta.__dict__.copy()
+            for name in self.meta.__dict__:
+                # Ignore any private attributes that Django doesn't care about.
+                # NOTE: We can't modify a dictionary's contents while looping
+                # over it, so we loop over the *original* dictionary instead.
+                if name.startswith('_'):
+                    del meta_attrs[name]
+            #from django.db.models.options import DEFAULT_NAMES
+            #for attr_name in DEFAULT_NAMES:
+            #    if attr_name in meta_attrs:
+            #        setattr(self, attr_name, meta_attrs.pop(attr_name))
+            #    elif hasattr(self.meta, attr_name):
+            #        setattr(self, attr_name, getattr(self.meta, attr_name))
+
+            # unique_together can be either a tuple of tuples, or a single
+            # tuple of two strings. Normalize it to a tuple of tuples, so that
+            # calling code can uniformly expect that.
+            ut = meta_attrs.pop('unique_together', self.unique_together)
+            if ut and not isinstance(ut[0], (tuple, list)):
+                ut = (ut,)
+            self.unique_together = ut
+
+            # Any leftover attributes must be invalid.
+            if meta_attrs != {}:
+                raise TypeError("'class Meta' got invalid attribute(s): %s" % ','.join(meta_attrs.keys()))
+        del self.meta
+
+    @property
+    def concrete_fields(self):
+        return self.fields
+
+    @property
+    def concrete_model(self):
+        return self.model
+
+    # make django filter happy.
+    def get_field_by_name(self, name):
+        for field in self.fields:
+            if field.attname == name:
+                return field
+        raise KeyError('No field named "{}" in model "{}".'.format(name, str(self.model)))
+
+    def setup_pk(self, field):
+        if not self.pk and field.primary_key:
+            self.pk = field
+
+    def add_field(self, field):
+        self.fields.append(field)
+        self.setup_pk(field)
+
```

Comment:
Remove one of the 3 blank lines

Reply:
I took care of all linting errors today.